xcode/Subconscious/Shared/Models/Petname.swift

```diff
@@ -7,6 +7,14 @@
 
 import Foundation
 
+public typealias TraversalPath = Petname?
+
+extension TraversalPath {
+    func concat(petname: Petname) -> Petname {
+        self?.concat(petname: petname) ?? petname
+    }
+}
```

Comment:
Open to suggestions on this abstraction, I found this helper very intuitive to work with.